bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcTable.scala

```diff
@@ -55,29 +67,78 @@ trait ItcTable extends Table {
     ConfigBridge.extractSequence(_, null, ConfigValMapInstances.IDENTITY_MAP, true)
   }
 
+  protected def calculateSpectroscopy(peer: Peer, c: ItcUniqueConfig): Future[ItcService.Result] =
+    Future {
+      List("Not Implemented Yet").fail
+    }
+
+  protected def calculateImaging(peer: Peer, ins: SPComponentType, c: ItcUniqueConfig): Future[ItcService.Result] =
+    ins match {
+      case INSTRUMENT_GMOS | INSTRUMENT_GMOSSOUTH => calculateImagingGmos(peer, ins, c)
+      case _                                      => Future { List("Not Implemented Yet").fail }
+    }
+
+  protected def calculateImagingGmos(peer: Peer, ins: SPComponentType, c: ItcUniqueConfig): Future[ItcService.Result] = {
+    val port = owner.getContextIssPort
+    //val wfs  = ??? TODO
+    val src  = new SourceDefinition(PointSource(20.0, BrightnessUnit.MAG), LibraryStar("A0V"), WavebandDefinition.R, 0.0)
+    val obs  = new ObservationDetails(ImagingSN(c.count, c.singleExposureTime, 1.0), AutoAperture(5.0))
+    val tele = new TelescopeDetails(TelescopeDetails.Coating.SILVER, port, Wfs.OIWFS)
+
+    val qual = owner.getContextSiteQuality
+    val cond = new ObservingConditions(qual.getImageQuality, qual.getCloudCover, qual.getWaterVapor, qual.getSkyBackground, 1.5)
+
+    // get the instrument configuration
+    val filter    = c.config.getItemValue(new ItemKey("instrument:filter")).asInstanceOf[GmosCommonType.Filter]
```

Comment:
If you're using this API a lot it might make sense to add some bedazzlement to force a checked cast, so you get an early failure rather than a mistyped value getting passed around.

```scala
implicit class ConfigOps(c: Config) {

  def get[A](key: ItemKey(implicit ev: ClassTag[A]): ClassCastException \/ A =
    \/.fromTryCatch(ev.runtimeClass.cast(a).asInstanceOf[A]).leftMap {
      case cce: ClassCastException => cce
      case ex => throw ex
    }

  def unsafeGet[A: ClassTag](key: ItemKey(implicit ev: ClassTag[A]): A =
    get[A](key).fold(throw _, identity)

}

// and then the cast is hidden
c.config.unsafeGet[GmosCommonType.Filter](new ItemKey("instrument:filter"))
```